great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1360,20 +1360,21 @@ def get_batch_data_and_markers(
         create_temp_table: bool = batch_spec.get(
             "create_temp_table", self._create_temp_table
         )
+        selectable: Union[
+            sqlalchemy.Selectable, str
+        ] = self._build_selectable_from_batch_spec(batch_spec=batch_spec)
+        # NOTE: what's being checked here is the presence of a `query` attribute, we could check this directly
+        # instead of doing an instance check
         if isinstance(batch_spec, RuntimeQueryBatchSpec):
             # query != None is already checked when RuntimeQueryBatchSpec is instantiated
-            query: str = batch_spec.query
 
             batch_data = SqlAlchemyBatchData(
                 execution_engine=self,
-                query=query,
+                selectable=selectable,
```

Comment:
This is what's new (compared to the `0.18.x` branch).
`_build_selectable_from_batch_spec` is what adds the splitter logic to the query.

Reply:
reverted this because it caused another problem